syftlib/src/main/java/org/openmined/syft/SignallingInterface.kt

```diff
@@ -0,0 +1,37 @@
+package org.openmined.syft
+
+import io.reactivex.Scheduler
+
+interface SignallingInterface {
+
+    /**
+     * computeThreadScheduler defines the thread on which observable runs
+     * @sample computeThreadScheduler Schedulers.io()
+     */
+    val computeThreadScheduler : Scheduler
```

Comment:
The schedulers should be defined in their own interface.
The will be used throughout the application and the library.

Reply:
Hmm I get it now. I was thinking of enforcing the onOpen/onClose messages to run on threads scheduled by these. Do you have suggestions for that?

Reply:
I think using `scheduler.scheduleDirect { callMeHere() }` should work.

Reply:
I've been thinking about the callbacks. Wouldn't be better to expose the Flowable and let the clients subscribe to it? This way we would have a reactive approach and won't need callbacks.

I have in my head a picture of the clients of this object (it could be the app or it could be a job) being accessed from a repository that would hide the implementation details leaving only the reactive function to hook on. But I'm digressing here :)

Reply:
Maybe we can go ahead with what we have so far and we can refactor after merging so that we can keep adding functionality.

Reply:
I think the procedure would go something like this then:

- We would need an interface for the subscriber that enforces the retry connection while leaving the rest for the user to implement

- In principle, they would still be implementing onMessage just inside the subscriber

-  This will however still ask the user to implement to onOpen and onClose callbacks for Socket which are independent of subscriber. These are embedded into observable logic.

Reply:
so only allowing for new subscribers doesn't give us benefits unless we need multiple threads running and hence multiple socket connections to PyGrid

Reply:
The main idea is to follow a paradigm (reactive) so that it's easier to know for anyone joining the project what is in our minds.
I think we can go as it is and refactor (if necessary) in the future.

Also there is a big chance that there will be more than one client at the same time: the UI and probably a job running in the background.

Reply:
Sure we can do that later!